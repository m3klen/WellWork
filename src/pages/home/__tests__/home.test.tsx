/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

import Home from '../'

describe('pages:home', () => {
  it('matches snapshot with default test props', async () => {
    const { container } = render(<Home />)

    expect(container).toMatchSnapshot()
  })

  it('uses the provided class name', async () => {
    const className = 'test-class'
    const { container } = render(<Home className={className} />)

    expect(container).toMatchSnapshot()
  })

  it('renders the template header text', async () => {
    const { container } = render(<Home />)

    expect(container).toMatchSnapshot()

    const header = container.querySelector('h2')

    expect(header).toHaveTextContent('React TypeScript Web App Template')
  })
})
