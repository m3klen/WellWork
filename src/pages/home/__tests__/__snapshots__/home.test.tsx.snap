// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages:home matches snapshot with default test props 1`] = `
<div>
  <div
    class="page-home"
  >
    <div
      class="page-home-content-wrapper"
    >
      <h2>
        React TypeScript Web App Template
      </h2>
    </div>
  </div>
</div>
`;

exports[`pages:home renders the template header text 1`] = `
<div>
  <div
    class="page-home"
  >
    <div
      class="page-home-content-wrapper"
    >
      <h2>
        React TypeScript Web App Template
      </h2>
    </div>
  </div>
</div>
`;

exports[`pages:home uses the provided class name 1`] = `
<div>
  <div
    class="page-home test-class"
  >
    <div
      class="page-home-content-wrapper"
    >
      <h2>
        React TypeScript Web App Template
      </h2>
    </div>
  </div>
</div>
`;
